command/agent/job_endpoint.go

```diff
@@ -140,13 +140,19 @@ func (s *HTTPServer) jobPlan(resp http.ResponseWriter, req *http.Request,
 		return nil, CodedError(400, "Job ID does not match")
 	}
 
+	// Backfill region from Job, if not present in WriteRequest
+	region := args.WriteRequest.Region
+	if region == "" {
+		region = *args.Job.Region
```

Comment:
should we protect against `Region` being nil here?  Does a job always have a Region field set?

Reply:
Just to be safe, we should probably guard against this.